### PR TITLE
Set 'indicate-multiline-delimiters=no' on default profile

### PR DESCRIPTION
**Pull request by gpetiot**
Setting the more "vanilla" value of this option for the default profile.
And also disabling the formatting of vendored libraries (should have been done long ago).

### CODE REVIEW

**Comment by Julow** [comment]
Why is this more "vanilla" ? Is there more discussion about this than https://github.com/ocaml-ppx/ocamlformat/pull/1377#issuecomment-682003304 ?

File: .ocamlformat-ignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+vendor/*/*
```
Comment (emillon) [inline .ocamlformat-ignore:1]: Can you pull this change in a different PR?

**Comment by emillon** [comment]
@Julow The two main arguments I would make for this feature to be opt-in:

- is that it's a style that's only possible with an automatic formatter. It's not the way most humans would write OCaml code. As such, it can be surprising seeing `_) )` somewhere.
- it is not super stable. that's not at the point of `doc-comments=after`, but we yanked changes to this feature from 0.15.0 because it was too impacting.

Both of these are at odds with the goals of the default profile, which is meant to be boring and stable in the long run.

File: .ocamlformat-ignore (hunk 1 above)
Comment (gpetiot) [reply .ocamlformat-ignore:1]: done in #1453

**Comment by Julow** [comment]
Seeing `) )` is indeed very surprising but I think this feature is useful for readability and while editing.
Setting it to `no` makes very compact code and makes it harder to select code that is deeply nested into parentheses, for example.

The most annoying example is nested matches:

```ocaml
let _ =
  match Foo.bar x with
  | This y -> (
    match Foo.baz y with
    | This z -> z
    | That w ->
        print something ;
        a (b (c w)))
  | That z -> z
```

I think the default profile has an other job, which is to ensure readability and avoid ambiguous syntaxes (to humans).
We should consider setting this option to `closing-on-separate-line`:

```ocaml
let _ =
  match Foo.bar x with
  | This y -> (
    match Foo.baz y with
    | This z -> z
    | That w ->
        print something ;
        a (b (c w))
  )
  | That z -> z
```

I agree that this option has and had a lot of bugs.
